spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/ReferenceLookupDelegate.java

```diff
@@ -116,7 +116,7 @@ public ReferenceLookupDelegate(
 		}
 
 		if (!result.iterator().hasNext()) {
-			return null;
+			return property.isMap() ? Collections.emptyMap() : null;
```

Comment:
This makes me wonder if we have the same problem with collections like `Set` and `List`. 
Care to add even more tests to double check?